File: lab2-functionalprimer/python3/func_as_arg.py
a=1
b=6
def listFunc():
    return [i for i in range(a, b)] #Create list of ints from 1 to 5, Haskell equivalent [1..5]

def applicatorFunc(inpFunc, s):
    if s=='s':
        return sum(inpFunc())
    else:
        return sum(inpFunc())/5

print(applicatorFunc(listFunc, 's'))

File: lab2-functionalprimer/haskell/func_as_arg.hs
--create inpFunc
-- a = 1
-- b = 10
-- inpFunc = [a..b] 

a = 1
b = 5
inpFunc s
    | s == 's' = [a..b]
    | s == 'a' = [a..b] 
    | otherwise = error "wrong"

-- inpFunc = [a..b] 

--Define applicatorFunc
applicatorFunc inpFunc s = if s=='s' then sum inpFunc else (sum inpFunc) / fromIntegral (length inpFunc)  

main = do
    let result = applicatorFunc inpFunc 's' 
    putStrLn("sum = " ++ show(result))
